cli/src/cli_util.rs

```diff
@@ -1625,6 +1626,14 @@ impl WorkspaceCommandTransaction<'_> {
         self.tx.mut_repo().edit(workspace_id, commit)
     }
 
+    pub fn rewrite_edited_commit(
```

Comment:
nit: it's unclear why `cmd_untrack()` doesn't use `.rewrite_edited_commit()`. Can you add some doc/inline comment?

I think it's also good to just migrate all CLI callers to `rewrite_edited_commit()`. Some of them might not be "edit", but I don't think it would practically matter.

Reply:
I added a comment in `cmd_untrack()`. The main issue is that since the workspace is locked, it's difficult to evaluate a revset here. I'm not sure `untrack` needs to reset the author information, since it seems different from the other commands already. I migrated all other CLI callers though.